server/verify.go

```diff
@@ -372,3 +376,48 @@ func VerifyGceTechnology(attestation *pb.Attestation, tech pb.GCEConfidentialTec
 	}
 	return fmt.Errorf("unknown GCEConfidentialTechnology: %v", tech)
 }
+
+// parseTCGEventLog is a wrapper function around `parsePCClientEventLog` method to:
+// 1. parse partial machine state from TPM TCG event logs.
+// 2. verify GceTechnology since the GCE Technology event is directly related to the TPM.
+func parseTCGEventLog(attestation *pb.Attestation, pcrs *tpmpb.PCRs, opts VerifyOpts) (*pb.MachineState, error) {
+	ms, err := parsePCClientEventLog(attestation.GetEventLog(), pcrs, opts.Loader)
+	if err != nil {
+		return nil, fmt.Errorf("failed to validate the PCClient event log: %w", err)
+	}
+
+	if err := verifyGceTechnology(attestation, ms.GetPlatform().Technology, &opts); err != nil {
+		return nil, fmt.Errorf("failed to verify memory encryption technology: %w", err)
+	}
+	return ms, nil
+}
+
+// parseTEEAttestation parses a machineState from TeeAttestation.
```

Comment:
The point of the wrapper func for parsePCClientEventlog is to avoid proto.Merge. proto.Merge is less readable since it implicitly copies every field from src to dst. This requires our implementation to be mutually exclusive, or we risk overwriting. We originally used proto.Merge to merge the state from two separate event logs, but we could just populate the TeeAttestation field directly in the wrapper.

Reply:
IOW, this can be inline in parseTCGEventLog.

Reply:
My take from "inline" is call `parseTEEAttestation` from `parseMachineStateFromTPM` after `verifyGceTechnolgy` is done. LMK if this is not what you want.